src/kitovu/sync/filecache.py

```diff
@@ -152,4 +154,8 @@ def discover_changes(self,
 
         remote_digest: str = plugin.create_remote_digest(remote_full_path)
         local_digest: str = plugin.create_local_digest(local_full_path)
+
+        # Case 5 if both change to the same values
+        if remote_digest == local_digest and file.cached_digest != remote_digest:
```

Comment:
I'm not sure if this is the correct approach, but it's too hot for me to think properly right now :sweat_smile:
Need to think a bit more about it - wouldn't it be better to only overwrite the cached digest if we know we don't have any (i.e. it's `''` or `None`)?

@bytinbit maybe you can take a look too?

Reply:
The comment should make this situation clear somehow:

![image](https://user-images.githubusercontent.com/625793/39768858-c5153f1e-52ea-11e8-8dcc-5b6f3b5bb7a7.png)